packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -1105,6 +1105,229 @@ describe('PhishingDetector', () => {
+            fuzzylist: [],
+            requestBlocklist: [],
+            name: 'test-config',
+            version: 1,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://example.com',
+          );
+          expect(result).toStrictEqual({
+            result: false,
+            type: 'requestBlocklist',
+          });
+        },
+      );
+    });
+
+    it('should call sha256Hash and check the hash against requestBlocklist, returning the correct result', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: [
+              '0415f1f12f07ddc4ef7e229da747c6c53a6a6474fbaf295a35d984ec0ece9455',
+            ],
+            name: 'test-config',
+            version: 1,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://develop.d3bkcslj57l47p.amplifyapp.com',
+          );
+          expect(result).toStrictEqual({
+            name: 'test-config',
+            result: true,
+            type: 'requestBlocklist',
+            version: '1',
+          });
+        },
+      );
+    });
+
+    it('should call sha256Hash and check the hash against requestBlocklist, returning the correct result without a version', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: [
+              '0415f1f12f07ddc4ef7e229da747c6c53a6a6474fbaf295a35d984ec0ece9455',
+            ],
+            name: 'test-config',
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain(
+            'https://develop.d3bkcslj57l47p.amplifyapp.com',
+          );
+          expect(result).toStrictEqual({
+            name: 'test-config',
+            result: true,
+            type: 'requestBlocklist',
+            version: undefined,
+          });
+        },
+      );
+    });
+
+    it('should return false if URL is invalid', async () => {
+      await withPhishingDetector(
+        [
+          {
+            blocklist: [],
+            fuzzylist: [],
+            requestBlocklist: [
+              '0415f1f12f07ddc4ef7e229da747c6c53a6a6474fbaf295a35d984ec0ece9455',
+            ],
+            name: 'test-config',
+            version: 1,
+            tolerance: 2,
+          },
+        ],
+        async ({ detector }) => {
+          const result = detector.isMaliciousRequestDomain('#$@(%&@#$(%');
+          expect(result).toStrictEqual({
+            result: false,
+            type: 'requestBlocklist',
+          });
+        },
+      );
+    });
+
+    it('should return true if URL is in the requestBlocklist', async () => {
```

Comment:
Test duplicate: should call sha256Hash and check the hash against requestBlocklist, returning the correct result